zebra-consensus/src/block/check.rs

```diff
@@ -15,28 +15,46 @@ use crate::{error::*, parameters::SLOW_START_INTERVAL};
 
 use super::subsidy;
 
-/// Returns `Ok(())` if there is exactly one coinbase transaction in `Block`,
-/// and that coinbase transaction is the first transaction in the block.
+/// Checks if there is exactly one coinbase transaction in `Block`,
+/// and if that coinbase transaction is the first transaction in the block.
+/// Returns the coinbase transaction is successful.
 ///
 /// "The first (and only the first) transaction in a block is a coinbase
 /// transaction, which collects and spends any miner subsidy and transaction
 /// fees paid by transactions included in this block." [§3.10][3.10]
 ///
 /// [3.10]: https://zips.z.cash/protocol/protocol.pdf#coinbasetransactions
-pub fn coinbase_is_first(block: &Block) -> Result<(), BlockError> {
+pub fn coinbase_is_first(block: &Block) -> Result<Arc<transaction::Transaction>, BlockError> {
```

Comment:
Optional: this name might be easier to understand, or we could use `get_coinbase`?

```suggestion
pub fn get_valid_coinbase(block: &Block) -> Result<Arc<transaction::Transaction>, BlockError> {
```

Reply:
I'm not sure, the main purpose is checking if the coinbase is the first tx, getting it is a side effect